input/kafkamdm/kafkamdm.go

```diff
@@ -221,6 +235,14 @@ func (k *KafkaMdm) consumePartition(topic string, partition int32, partitionOffs
 			if err := offsetMgr.Commit(topic, partition, currentOffset); err != nil {
 				log.Error(3, "kafka-mdm failed to commit offset for %s:%d, %s", topic, partition, err)
 			}
+			partitionOffset[partition].Set(int(currentOffset))
```

Comment:
I suggest we do the map lookups at the beginning of the function, so we don't have to do it over and over again in the loop.
And actually, a list seems like a more effective type than a map since partitions start at 0 and are ascending.
apparently there's no issue using int32's as a list index https://play.golang.org/p/eXqRml8o8h